site/content/features/virtual-threads.md

```diff
@@ -0,0 +1,433 @@
+
+There are a few differences. In particular, all virtual threads:
+
+* Are in a single thread group 
+* Have priority `NORM_PRIORITY`
+* Are daemon threads
+
+There is no API for constructing a virtual thread with another thread group. Trying to call `setPriority` or `setDaemon` on a virtual thread has no effect.
+
+The static `Thread::getAllStackTraces` method returns a map of stack traces of all *platform* threads. Virtual threads are not included.
+
+A new `Thread::isVirtual` instance method tells whether a thread is virtual.
+
+Note that there is no way to find the platform thread on which a virtual thread executes.
+
+Java 19 has a couple of changes to the `Thread` API that have nothing to do with virtual threads:
+
+* There are now instance methods `join(Duration)` and `sleep(Duration)`.
+* The non-final `getId` method is deprecated since someone might override it to return something other than the thread ID. Call the final `threadId` method instead.
+
+As of Java 20, the `stop`, `suspend`, `resume`, `countStackFrames` methods throw an `UnsupportedOperationException` for both platform and virtual threads. These methods have been deprecated since Java 1.2 and deprecated for removal since Java 18.
+
+## Capturing Task Results
+
+You often want to combine the results of multiple concurrent tasks:
+
+```
+Future<T1> f1 = service.submit(callable1);
+Future<T2> f2 = service.submit(callable2);
+result = combine(f1.get(), f2.get());
+```
+
+Before virtual threads, you might have felt bad about the blocking `get` calls. But now blocking is cheap. Here is a sandbox with a more concrete example:
+
+{{< sandbox version=java21 preview="true" mainclass="VirtualThreadDemo" >}}{{< sandboxsource "VirtualThreadDemo.java" >}}
+import java.util.concurrent.*;
+import java.net.*;
+import java.net.http.*;
+
+public class VirtualThreadDemo {
+   public static void main(String[] args) throws InterruptedException, ExecutionException {
+      ExecutorService service = Executors.newVirtualThreadPerTaskExecutor();
+      Future<String> f1 = service.submit(() -> get("https://horstmann.com/random/adjective"));
+      Future<String> f2 = service.submit(() -> get("https://horstmann.com/random/noun"));
+      String result = f1.get() + " " + f2.get();
+      System.out.println(result);
+      service.close();
+   }
+
+   private static HttpClient client = HttpClient.newHttpClient();
+
+   public static String get(String url) {
+      try {
+         var request = HttpRequest.newBuilder().uri(new URI(url)).GET().build();
+         return client.send(request, HttpResponse.BodyHandlers.ofString()).body();
+      } catch (Exception ex) {
+         throw new RuntimeException(ex);
+      }
+   }   
```

Comment:
As you're using the "callable trick" probably no exception handler is required at all:

```suggestion
   public static String get(String url) throws Exception {
      var request = HttpRequest.newBuilder().uri(new URI(url)).GET().build();
      return client.send(request, HttpResponse.BodyHandlers.ofString()).body();
   }   
```

Reply:
Changed to `public static String get(String url) throws InterruptedException, IOException, URISyntaxException`